### ISSUE TITLE
TODO & FIXME

### ISSUE THREAD

**Issue by lfaoro** [realm/SwiftLint#150]
Why does the linter checks for TODO/FIXME statements? I thought you guys are using GitHub's Swift Style Guide as a guideline which has all rules explained.

If you add rules of your own, please document the reasoning behind it.

**Comment by keith** [comment]
If you'd like to disable this for your project you can add this to your `.swiftlint.yml`

```yml
disabled_rules:
  - todo
```

**Comment by keith** [comment]
Also the readme says:

> loosely based on GitHub's Swift Style Guide

So I don't think it's a huge deal as more rules are added to SwiftLint. Especially since they can be easily disabled.

**Comment by lfaoro** [comment]
I know how to disable it, thank you.
What I'd like to know is: Where is the reasoning behind this decision documented.

**Comment by jpsim** [comment]
> What I'd like to know is: Where is the reasoning behind this decision documented?

I added that in because IMO the role of a linter is to show you the parts of your code that need your attention and ultimately need to be fixed. TODOs and FIXMEs are one way to do this.

**Comment by donileo** [comment]
@jpsim I'm not sure I understand. Are you saying the reason for the warning is to actually show and bring the todo to your attention so that you may implement the actual todo?

**Comment by jpsim** [comment]
Yes.

**Comment by donileo** [comment]
@jpsim  The message given by Swift lint does not imply that though: 

SwiftLint displays: "Todo Violation: TODOs should be avoided...."

**Comment by jpsim** [comment]
If you have an improvement in mind, feel free to file a PR.

**Comment by donileo** [comment]
@jpsill Will do!

**Comment by sbohmann** [comment]
I first also thought that it discourages using TODOs but it actually encourages them by supporting them and making them more useful through this rule :D Maybe the message should be changed to "consider doing / fixing what the TODO suggests", though ^^

**Comment by donileo** [comment]
Pull Request [2069](https://github.com/realm/SwiftLint/pull/2069) has been added to fix this issue.

**Comment by Andy0570** [comment]
The author's intent in adding `todo` to lint is that code or functionality annotated via TODO or FIXME should be refined or implemented, so by way of a warning you are reminded that there is unfinished functionality in your project. 

If you want to suppress this warning, you can turn off this rule, which is on by default, in `disabled_rules`, and you can also use 'notaTODO' and 'notaFIXME' instead;
